### PR TITLE
Cite individual corpora

### LINKED ISSUE #18: Cite constituent corpora
Individual corpora aggregated in DoReCo should be cited in
- `README.md`
- `.zenodo.json`
- CLDF metadata

### PR DESCRIPTION
**Pull request by xrotwang**
closes #18

### CODE REVIEW

File: cldfbench_doreco.py [hunk 1]
```diff
@@ -112,6 +119,23 @@ def cmd_download(self, args):
...
+            "the name(s) of the creator(s) of each corpus as given below."]
+        zenodo['description'] += '\n<p>{}</p>\n'.format(corpus_citations[0])
+        for src in self.cldf_reader().properties['prov:wasDerivedFrom']:
+            if src.get('dc:title', '').endswith('DoReCo dataset'):
+                corpus_citations.append('> {}\n'.format(src['dc:bibliographicCitation']))
+                zenodo['description'] += '\n<blockquote>{}</blockquote>\n'.format(
+                    html.escape(src['dc:bibliographicCitation']))
+        dump(zenodo, self.dir / '.zenodo.json', indent=4)
+        pre, head, post = super().cmd_readme(args).partition('## Description')
+        md = pre + '\n'.join(corpus_citations) + '\n\n' + head + post
```
Comment (xrotwang) [inline cldfbench_doreco.py:137]: Here we add the citations to README.md and .zenodo.json

File: cldfbench_doreco.py [hunk 2]
```diff
@@ -191,6 +215,14 @@ def cmd_makecldf(self, args):
         for row in self.raw_dir.read_csv('languages.csv', dicts=True):
             if not self.raw_dir.joinpath('{}_metadata.csv'.format(row['Glottocode'])).exists():
                 continue
+
+            args.writer.cldf.add_provenance(wasDerivedFrom={
+                "rdf:about": "https://github.com/cldf-datasets/doreco/",
+                "rdf:type": "prov:Entity",
+                "dc:title": "{} DoReCo dataset".format(row['Language']),
+                "dc:bibliographicCitation": CORPUS_CITATION_FMT.format(**row)
+            })
```
Comment (xrotwang) [inline cldfbench_doreco.py:224]: and here to the CLDF JSON metadata.

**Review by FredericBlum — APPROVED** [review]
Cool! Thanks for implementing this

**Comment by FredericBlum** [comment]
I am confused about this. We have merged this, but the README on the GitHub page is still unchanged. Since this is where most people will probably look at, shouldn't it appear there as well? I thought I had seen it actually, but it does not seem to be featured right now.

**Comment by xrotwang** [comment]
I haven't added/updated any data-related stuff yet, because we worked with the set including the NC licensed corpora most of the time. For the doreco CLDF release we'd exclude these, so the README (and data) would change to reflect that.

So, after you extract the exact dataset for the study, we could recreate the CLDF data with the smaller set of corpora and then release it with the README reflecting this reduced set.

**Comment by xrotwang** [comment]
If you want the citations of all corpora for the study, you could just run `cldfbench readme cldfbench_doreco.py` locally and then inspect the README.

**Comment by xrotwang** [comment]
> If you want the citations of all corpora for the study, you could just run `cldfbench readme cldfbench_doreco.py` locally and then inspect the README.

It might be worth adding this to the usage notes to make sure people who use the NC corpora as well will also not forget to cite them.

**Pull request merged**